src/main/java/lotto/domain/LottoResult.java

```diff
@@ -5,29 +5,26 @@
 
 public class LottoResult {
 
-    private final Map<Rank, Integer> ranks = new LinkedHashMap<>();
+    private final Map<Rank, Integer> lottoResult;
 
-    public LottoResult() {
-        initRanks();
-    }
-
-    private void initRanks() {
+    public LottoResult(Map<Rank, Integer> ranks) {
+        Map<Rank, Integer> result = new LinkedHashMap<>();
         for (int i = 1; i < Rank.values().length; i++) {
-            ranks.put(Rank.values()[i], 0);
+            result.put(Rank.values()[i], 0);
         }
+        for (Map.Entry<Rank, Integer> rank : ranks.entrySet()) {
+            result.put(rank.getKey(), rank.getValue());
+        }
+        this.lottoResult = result;
     }
 
-    public Map<Rank, Integer> matchRank(WinningLotto winningLotto, LottoGroup lottos) {
-        for (Lotto lotto : lottos.getLottoGroup()) {
-            Rank rank = winningLotto.findRank(lotto);
-            ranks.computeIfPresent(rank, (key, val) -> val + 1);
-        }
-        return ranks;
+    public Map<Rank, Integer> getLottoResult() {
+        return lottoResult;
```

Comment:
```suggestion
    public Map<Rank, Integer> getLottoResult() {
        return Collections.unmodifiableMap(lottoResult);
```
unmodifiableMap도 있어요!